# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1229980587 -3600
#      Mon Dec 22 22:16:27 2008 +0100
# Node ID fa12c67683d34911e794834714d572867ec04e71
# Parent  36c8a3696ae746215326d324d800db65326fbf46
Reszie third arg of __voronoi__ before retuning it

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-12-22  David Bateman  <dbateman@free.fr>
+
+	* DLD-FUNCTIONS/__voronoi__.cc (F__voronoi__): Resize AtInf array
+	before returning it.
+
 2008-12-12  David Bateman  <dbateman@free.fr>
 
 	* DLD-FUNCTIONS/chol.cc (Fcholinv): Add test.
 
 2008-12-12  Jaroslav Hajek  <highegg@gmail.com>
 
 	* ls-hdf5.cc (add_hdf5_data): Check for diagonal & permutation
 	matrices and convert them to full prior to saving.
diff --git a/src/DLD-FUNCTIONS/__voronoi__.cc b/src/DLD-FUNCTIONS/__voronoi__.cc
--- a/src/DLD-FUNCTIONS/__voronoi__.cc
+++ b/src/DLD-FUNCTIONS/__voronoi__.cc
@@ -160,17 +160,17 @@ Undocumented internal function.\n\
 
       Matrix v (n, dim);
       for (octave_idx_type d = 0; d < dim; d++)
 	v(0,d) = octave_Inf;
 
       boolMatrix AtInf (np, 1);
       for (i = 0; i < np; i++) 
 	AtInf(i) = false;
-      octave_value_list F;
+      octave_value_list F (np, octave_value ());
       k = 0;
       i = 0;
 
       FORALLfacets 
 	{
 	  facet->seen = false;
 	}
 
@@ -215,16 +215,17 @@ Undocumented internal function.\n\
 	}
 
       Cell C(r, 1);
       for (i = 0; i < r; i++)
 	C.elem (i) = F(i);
 
       retval(0) = v;
       retval(1) = C;
+      AtInf.resize (r, 1);
       retval(2) = AtInf;
 
       // free long memory
       qh_freeqhull (! qh_ALL);
 
       // free short memory and memory allocator
       int curlong, totlong;
       qh_memfreeshort (&curlong, &totlong);
